packages/dds/register-collection/src/consensusRegisterCollection.ts

```diff
@@ -142,18 +142,19 @@ export class ConsensusRegisterCollection<T>
 	 * @returns Promise<true> if write was non-concurrent
 	 */
 	public async write(key: string, value: T): Promise<boolean> {
-		const serializedValue = this.stringify(value, this.serializer);
-
 		if (!this.isAttached()) {
 			// JSON-roundtrip value for local writes to match the behavior of going through the wire
-			this.processInboundWrite(key, this.parse(serializedValue, this.serializer), 0, 0, true);
+			this.processInboundWrite(key, value, 0, 0, true);
 			return true;
 		}
 
-		const message: IRegisterOperation = {
+		const message: IRegisterOperationOld<T> = {
```

Comment:
Rename this type to not say `Old` anymore?

Reply:
And likewise rename `incomingOpMatchesCurrentFormat`, it's backwards now according to that name